src/containers/Dev/Snapshots.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { useState, useEffect, useRef } from 'react'
+import {
+  Button,
+  Grid,
+  GridColumn,
+  Icon,
+  Input,
+  Label,
+  Loader,
+  Popup,
+  Portal,
+  Segment,
+} from 'semantic-ui-react'
+import config from '../../config'
+
+const snapshotsBaseUrl = `${config.serverREST}/snapshot`
+const snapshotListUrl = `${snapshotsBaseUrl}/list`
+const takeSnapshotUrl = `${snapshotsBaseUrl}/take`
+const useSnapshotUrl = `${snapshotsBaseUrl}/use`
+const snapshotFilesUrl = `${snapshotsBaseUrl}/files`
+const uploadSnapshotUrl = `${snapshotsBaseUrl}/upload`
+// const diffSnapshotUrl = `${snapshotsBaseUrl}/diff`
+
+const Snapshots: React.FC = () => {
+  const [isOpen, setIsOpen] = useState(false)
+  const [compareFrom, setCompareFrom] = useState('')
+  const [isPortalOpen, setIsPortalOpen] = useState(false)
+  const [isSnapshotError, setIsSnapshotError] = useState(false)
+
+  const [data, setData] = useState<string[] | null>(null)
+
+  useEffect(() => {
+    if (isOpen) {
+      setData(null)
+      setCompareFrom('')
+      setIsSnapshotError(false)
+      getList()
+    }
+  }, [isOpen])
+
+  const getList = async () => {
+    try {
+      const snapshotListRaw = await fetch(snapshotListUrl, { method: 'GET' })
+      const snapshotList: string[] = (await snapshotListRaw.json()).snapshotsNames
+
+      setData(snapshotList)
+    } catch (e) {}
+  }
+
+  const normaliseSnapshotName = (name: string) =>
+    // not word, not digit
+    name.replace(/[^\w\d]/g, '')
+
+  const takeSnapshot = async (name: string) => {
+    if (!name) return
+    setIsOpen(false)
+    setIsPortalOpen(true)
+    try {
+      const resultRaw = await fetch(`${takeSnapshotUrl}?name=${normaliseSnapshotName(name)}`, {
+        method: 'POST',
+      })
+      const resultJson = await resultRaw.json()
+
+      if (resultJson.success) return setIsPortalOpen(false)
+
+      setIsSnapshotError(true)
+    } catch (e) {
+      setIsSnapshotError(true)
+    }
```

Comment:
If there is an error, or jsonResult.success === false, will the Portal ever close?

Reply:
had to manually click on the close button in the portal error, now it's changed to be a modal, should close as per normal modal closing (or when clicking on 'X' in error label)